An error is happened when compiling the Libnd4j in Windows

I met the following error when compiling the libnd4j dll in Windows with msys2 and cmake (version 3.26.2):

G:/msys2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe:
 CMakeFiles/nd4jcpu.dir/objects.a(LegacyReduce3Op.cpp.obj):LegacyReduce3Op.cpp:(.text+0xcc5): undefi
ned reference to `sd::ops::LegacyReduce3Op::traceExecIfNeeded(sd::graph::Context&)'
collect2.exe: error: ld returned 1 exit status
make[2]: *** [blas/CMakeFiles/nd4jcpu.dir/build.make:2228: blas/libnd4jcpu.dll] Error 1

![image](https://user-images.githubusercontent.com/62918515/230718720-3c06f53c-1209-4404-b81c-e0d49879fdb4.png)

How can i solve it?

--- a/libnd4j/include/ops/declarable/LegacyReduce3Op.h
+++ b/libnd4j/include/ops/declarable/LegacyReduce3Op.h
@@ -39,7 +39,6 @@ class SD_LIB_EXPORT LegacyReduce3Op : public LegacyOp {
 
   ShapeList* calculateOutputShape(ShapeList* inputShape, sd::graph::Context& block) override;
   LegacyOp* clone() override;
-  void traceExecIfNeeded(Context& block);
 };
 }  // namespace ops
 }  // namespace sd

--- a/nd4j/nd4j-backends/nd4j-backend-impls/nd4j-cpu-backend-common/src/main/java/org/nd4j/linalg/cpu/nativecpu/ops/CpuOpContext.java
+++ b/nd4j/nd4j-backends/nd4j-backend-impls/nd4j-cpu-backend-common/src/main/java/org/nd4j/linalg/cpu/nativecpu/ops/CpuOpContext.java
@@ -64,6 +64,8 @@ public class CpuOpContext extends BaseOpContext implements OpContext, Deallocata
     @Override
     public void close() {
         // no-op
+        nativeOps.ctxPurge(context);
+        context.deallocate();
     }
 
     @Override
@@ -203,7 +205,7 @@ public class CpuOpContext extends BaseOpContext implements OpContext, Deallocata
         List<DataBuffer> shapeInfoReferences = new ArrayList<>();
         for(int i = 0; i < arrays.size(); i++) {
             INDArray array = arrays.get(i);
-            buffers.put(i,array.isEmpty() ? null : ((BaseCpuDataBuffer) array.data()).getOpaqueDataBuffer());
+            buffers.put(i,Carray.isEmpty() ? null : ((BaseCpuDataBuffer) array.data()).getOpaqueDataBuffer());
             DataBuffer dataBuffer = array.shapeInfoDataBuffer();
             shapeInfoReferences.add(dataBuffer);
             Pointer addressPointer = dataBuffer.pointer();

--- a/nd4j/nd4j-backends/nd4j-backend-impls/nd4j-cpu-backend-common/src/main/java/org/nd4j/linalg/cpu/nativecpu/ops/CpuOpContext.java
+++ b/nd4j/nd4j-backends/nd4j-backend-impls/nd4j-cpu-backend-common/src/main/java/org/nd4j/linalg/cpu/nativecpu/ops/CpuOpContext.java
@@ -205,7 +205,7 @@ public class CpuOpContext extends BaseOpContext implements OpContext, Deallocata
         List<DataBuffer> shapeInfoReferences = new ArrayList<>();
         for(int i = 0; i < arrays.size(); i++) {
             INDArray array = arrays.get(i);
-            buffers.put(i,Carray.isEmpty() ? null : ((BaseCpuDataBuffer) array.data()).getOpaqueDataBuffer());
+            buffers.put(i,array.isEmpty() ? null : ((BaseCpuDataBuffer) array.data()).getOpaqueDataBuffer());
             DataBuffer dataBuffer = array.shapeInfoDataBuffer();
             shapeInfoReferences.add(dataBuffer);
             Pointer addressPointer = dataBuffer.pointer();

--- a/nd4j/nd4j-backends/nd4j-backend-impls/nd4j-cuda/src/main/java/org/nd4j/linalg/jcublas/ops/executioner/CudaOpContext.java
+++ b/nd4j/nd4j-backends/nd4j-backend-impls/nd4j-cuda/src/main/java/org/nd4j/linalg/jcublas/ops/executioner/CudaOpContext.java
@@ -62,6 +62,8 @@ public class CudaOpContext extends BaseOpContext implements OpContext, Deallocat
 
     @Override
     public void close() {
+        nativeOps.ctxPurge(context);
+        context.deallocate();
     }
 
     @Override

